Shopify/ruby-lsp
# Migrate PathCompletion to use the index

### Motivation

Closes #938

Start providing require path autocomplete through the index instead of re-building the prefix tree on every request.

### Implementation

Exposed a way to search for require paths from the index and started using it in path completion.

### Automated Tests

Adapted our existing tests.

### Manual Tests

1. Start the LSP on this branch
2. Type a require
3. Verify you get correct completion after insert the forward slash
